src/components/tl/ucp/tl_ucp.c

```diff
@@ -175,6 +175,20 @@ static ucs_config_field_t ucc_tl_ucp_context_config_table[] = {
      ucc_offsetof(ucc_tl_ucp_context_config_t, pre_reg_mem),
      UCC_CONFIG_TYPE_UINT},
 
+    {"SERVICE_TLS", "",
+     "If empty string (default), uses the same worker for collectives and "
+     "service. If not, creates a special worker for service collectives "
+     "for which UCX_TL configuration is set to that string",
+     ucc_offsetof(ucc_tl_ucp_context_config_t, service_tls),
+     UCC_CONFIG_TYPE_STRING},
+
+    {"SERVICE_NET_DEVICES", "",
```

Comment:
i suggest we add another variable that explicitly enables alternative worker for service. Currently, this flow is triggered if SERVICE_TLS is set. Suppose, user wants a dedicated NET iface for service but default set of TLS (user might not know anything about UCX TLS). Then only setting SERVICE_NET_DEVICES would not work. Lets add "UCC_TL_UCP_SERVICE_WORKER=y/n". If that var is set we use the flow. SERVICE_TLS and SERVICE_NET_DEVICES are parsed as is. @kingchc wdyt?

Reply:
Makes sense to be explicit. Can this be documented somewhere?

Reply:
Ok it makes sense. I implement it in a new separated commit.
About the documentation @vspetrov what do you suggest?

Reply:
> Ok it makes sense. I implement it in a new separated commit. About the documentation @vspetrov what do you suggest?

Parameter description will be available via ucc_info. Also let's add Wiki FAQ page: "custom service ucp worker"